### ISSUE TITLE
Type check string % operator

### ISSUE THREAD

**Issue by JukkaL** [python/mypy#469]
We should flag cases like these as errors:

```
print('%d %d' % (1, 2, 3))   # Too many items in the right operand
print('%d' % 'foo')   # A number is required not str
```

This is actually pretty important, since `%` is often used in exception handlers to create exception error messages, and exception handlers are often difficult to test.
